ERROR: AssertionError: is_leaf(frame)

I'm trying to run Debugger on Pkg using the usual trick of commenting out my dev Pkg's uuid in the project file and then loading that one as a new package and then running it in the debugger and I'm getting this error:
```jl
julia> @enter Pkg.resolve()
resolve() in Pkg.API at /Users/stefan/dev/Pkg/src/API.jl:172, pc = 1
  Pkg.Types.Context() in Pkg.Types at util.jl:723
    Pkg.Types.EnvCache() in Pkg.Types at /Users/stefan/dev/Pkg/src/Types.jl:314
      Pkg.Types.EnvCache(env::Union{Nothing, String}) in Pkg.Types at /Users/stefan/dev/Pkg/src/Types.jl:314
        find_project_file(env::Union{Nothing, String}) in Pkg.Types at /Users/stefan/dev/Pkg/src/Types.jl:207
          safe_realpath(path) in Pkg.Types at /Users/stefan/dev/Pkg/src/Types.jl:480
            realpath(path::AbstractString) in Base.Filesystem at path.jl:375
              getproperty(x::Module, f::Symbol) in Base at Base.jl:13
node = :(($(QuoteNode(Pkg.Types.Context)))())
ERROR: AssertionError: is_leaf(frame)
Stacktrace:
 [1] step_expr!(::Any, ::JuliaInterpreter.Frame, ::Any, ::Bool) at /Users/stefan/.julia/packages/JuliaInterpreter/8bMSs/src/interpret.jl:404
 [2] step_expr! at /Users/stefan/.julia/packages/JuliaInterpreter/8bMSs/src/interpret.jl:536 [inlined]
 [3] next_until!(::Any, ::Any, ::JuliaInterpreter.Frame, ::Bool) at /Users/stefan/.julia/packages/JuliaInterpreter/8bMSs/src/commands.jl:93
 [4] next_call! at /Users/stefan/.julia/packages/JuliaInterpreter/8bMSs/src/commands.jl:112 [inlined]
 [5] maybe_step_through_wrapper!(::Any, ::JuliaInterpreter.Frame) at /Users/stefan/.julia/packages/JuliaInterpreter/8bMSs/src/commands.jl:246
 [6] maybe_step_through_wrapper!(::JuliaInterpreter.Frame) at /Users/stefan/.julia/packages/JuliaInterpreter/8bMSs/src/commands.jl:256
 [7] top-level scope at /Users/stefan/.julia/packages/Debugger/JwW1u/src/Debugger.jl:90
```
I have also set a breakpoint, which seems to be necessary for this to be triggered. This is the breakpoint I set: `@breakpoint Pkg.Operations.resolve_versions!`.

Repro with JuliaInterpreter:

```
julia> frame = JuliaInterpreter.enter_call(Pkg.resolve)
Frame for resolve() in Pkg.API at C:\Users\Kristoffer\AppData\Local\Julia-1.1.0\share\julia\stdlib\v1.1\Pkg\src\API.jl:123
  1 123  1 ─ %1 = (Pkg.Types.Context)()
  2 123  │   %2 = (#self#)(%1)
  3 123  └──      return %2

julia> JuliaInterpreter.maybe_step_through_wrapper!(frame)
resolve() in Pkg.API at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Pkg\src\API.jl:123, pc = 1
  Pkg.Types.Context() in Pkg.Types at util.jl:728
    Pkg.Types.EnvCache() in Pkg.Types at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Pkg\src\Types.jl:278
      Pkg.Types.EnvCache(env::Union{Nothing, String}) in Pkg.Types at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Pkg\src\Types.jl:278
        read_project(path::String) in Pkg.Types at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Pkg\src\Types.jl:495
          open(f::Function, args...) in Base at iostream.jl:367
            #open#310(kwargs, ::Any, f::Function, args...) in Base at iostream.jl:367
              (::getfield(Pkg.Types, Symbol("##44#45")))(io) in Pkg.Types at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Pkg\src\Types.jl:496
                (::getfield(Pkg.Types, Symbol("#kw##read_project")))(::Any, ::typeof(Pkg.Types.read_project), io::IO) in Pkg.Types
                  #read_project#41(path, ::Any, io::IO) in Pkg.Types at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Pkg\src\Types.jl:480
                    getproperty(x::Module, f::Symbol) in Base at sysimg.jl:13
node = :(($(QuoteNode(Pkg.Types.Context)))())
ERROR: AssertionError: is_leaf(frame)
Stacktrace:
 [1] step_expr!(::Any, ::Frame, ::Any, ::Bool) at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\interpret.jl:404
 [2] step_expr! at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\interpret.jl:536 [inlined]
 [3] next_until!(::Any, ::Any, ::Frame, ::Bool) at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\commands.jl:93
 [4] next_call! at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\commands.jl:112 [inlined]
 [5] maybe_step_through_wrapper!(::Any, ::Frame) at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\commands.jl:246
 [6] maybe_step_through_wrapper!(::Frame) at C:\Users\Kristoffer\Debugging\JuliaInterpreter\src\commands.jl:256
 [7] top-level scope at none:0
```

--- a/src/commands.jl
+++ b/src/commands.jl
@@ -244,7 +244,9 @@ function maybe_step_through_wrapper!(@nospecialize(recurse), frame::Frame)
         # this is a wrapper function that we might want to step through
         while frame.pc != length(stmts)-1
             pc = next_call!(recurse, frame, false)  # since we're in a Method we're not at toplevel
-            pc === nothing && return frame
+            if pc === nothing || isa(pc, BreakpointRef)
+                return frame
+            end
         end
         ret = evaluate_call!(dummy_breakpoint, frame, last)
         @assert isa(ret, BreakpointRef)

--- a/test/debug.jl
+++ b/test/debug.jl
@@ -432,6 +432,7 @@ struct B{T} end
         frame = JuliaInterpreter.enter_call(f)
         JuliaInterpreter.maybe_step_through_wrapper!(frame)
         @test leaf(frame).framecode.scope == @which g()
+    end
 
     @testset "preservation of stack when throwing to toplevel" begin
         f() = "αβ"[2]

--- a/test/debug.jl
+++ b/test/debug.jl
@@ -424,4 +424,13 @@ struct B{T} end
         frame, pc = debug_command(frame, :si)
         @test frame.pc == 1
     end
+
+    @testset "breakpoints hit during wrapper step through" begin
+        f(x = g()) = x
+        g() = 5
+        @breakpoint g()
+        frame = JuliaInterpreter.enter_call(f)
+        JuliaInterpreter.maybe_step_through_wrapper!(frame)
+        @test leaf(frame).framecode.scope == @which g()
+    end
 # end

